### PR TITLE
ISPN-9715 Update internal embedded listeners to be non blocking

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-9715

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java [hunk 1]
```diff
@@ -90,6 +90,7 @@ private void updateSearchFactory(final Class<?>... classes) {
          if (reducedSet.isEmpty()) {
             return;
          }
+         // TODO: may want to make this run on a different thread to make non blocking when invoking addClasses?
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java]: I need input from someone who is more familiar with query if I need to move the execution of `addClasses` to another thread? It looks like it is probably a blocking operation but I couldn't quite tell.

File: query/src/main/java/org/infinispan/query/indexmanager/ClusteredSwitchingBackend.java [hunk 2]
```diff
@@ -116,6 +116,7 @@ public void viewChanged(final ViewChangedEvent e) {
       final int viewId = e.getViewId();
       if (viewId > currentViewId) {
          if (lastSeenViewId.compareAndSet(currentViewId, viewId)) {
+            // TODO: it looks like this can be blocking - may need to offload
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/indexmanager/ClusteredSwitchingBackend.java]: I need input from someone who is more familiar with query if I need to move this to another thread.

Comment (tristantarrant) [reply query/src/main/java/org/infinispan/query/indexmanager/ClusteredSwitchingBackend.java]: It definitely can be blocking:
* if this was the master and it has to relinquish control, it needs to flush the index and close
* if this is becoming the master, it needs to reopen the index in rw mode

Comment (wburns) [reply query/src/main/java/org/infinispan/query/indexmanager/ClusteredSwitchingBackend.java]: Okay, I will fix this then.

**Comment by wburns** [comment]
Also have to look into some test failures here https://ci.infinispan.org/job/Infinispan/job/PR-6979/1/

File: query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java (hunk 1 above)
Comment (wburns) [reply query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java]: Going to assume this is blocking as well and change it.

Comment (danberindei) [reply query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java]: The MutableSearchFactory impl uses a lock, so I'd also say it's blocking /cc @gustavonalle @anistor

Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java]: yes, we talked about it yesterday

Comment (wburns) [reply query/src/main/java/org/infinispan/query/backend/SearchFactoryHandler.java]: This should be handled in https://github.com/infinispan/infinispan/pull/7010

**Pull request merged**